packages/core-contracts/test/utils/Ownable.test.js

```diff
@@ -0,0 +1,196 @@
+const { ethers } = hre;
+const assert = require('assert/strict');
+
+const { ZERO_ADDRESS } = require('../test-libs/test-lib');
+
+describe('Ownable', () => {
+  describe('Constructor / Deployment', () => {
+    it('Should revert when no argument passed to Ctor', async () => {
+      const OwnableParent = await ethers.getContractFactory('OwnableParent');
+
+      await assert.rejects(
+        async () => {
+          await OwnableParent.deploy();
+        },
+        {
+          name: 'Error',
+          message:
+            `missing argument:  in Contract constructor (count=0,` +
+            ` expectedCount=1, code=MISSING_ARGUMENT, version=contracts/5.4.1)`,
+        }
+      );
+    });
+
+    it('Should revert when owner parameter is passed the zero address', async () => {
+      const OwnableParent = await ethers.getContractFactory('OwnableParent');
+
+      await assert.rejects(
+        async () => {
+          await OwnableParent.deploy(ZERO_ADDRESS);
+        },
+        {
+          name: 'Error',
+          message:
+            `VM Exception while processing transaction: reverted with` +
+            ` reason string 'Owner address cannot be 0x0'`,
+        }
+      );
+    });
+
+    it('Should set owner address on deployment', async () => {
```

Comment:
As described above, the pattern we are trying to follow is to deploy only once and operate on that instance, vs deploying on each test. Our `it` cases do only one thing. The whole test spreads out like a tree and traverses all possible branches.

Reply:
I'll mark this as resolved as it is a duplicate of the comment above that you mention: https://github.com/Synthetixio/synthetix-v3/pull/187#discussion_r718766820